### PR TITLE
Added gstreamer support to the magicleap port

### PR DESCRIPTION
**Pull request by asajeffrey**
With this PR we can watch videos in magicleap servo.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes do not require tests because I'm not sure how we'd test it

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23929)

### CODE REVIEW

**Comment by asajeffrey** [comment]
Gstreamer needs the homebrew bison, not the system one.

@bors-servo try=magicleap

**Comment by asajeffrey** [comment]
Argh, it was still picking up the system bison, I suspect because the env isn't being passed through, including the custom `$PATH`. @bors-servo try=magicleap

**Comment by asajeffrey** [comment]
Well we got further this time... My guess is that the CFLAGS being set by mach and there ones being set by meson aren't playing well together.

**Comment by asajeffrey** [comment]
Ah, also the SDK version is `v0.20.0+ndk19c` which isn't going to work anyway.

**Comment by asajeffrey** [comment]
Added the meson log file as a build artifact. @bors-servo try=magicleap

**Comment by asajeffrey** [comment]
```
/aarch64-linux-android/bin/ld: cannot find crtend_android.o: No such file or directory
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by asajeffrey** [comment]
Got mach to download a prebuilt gstreamer from s3. Added the script to build the tarball to support/magicleap. @bors-servo try=magicleap

**Comment by asajeffrey** [comment]
Ah, that build succeeded! The failure is due to the missing meson log artifact.

**Comment by asajeffrey** [comment]
Removed the meson log artefact. @bors-servo try=magicleap

**Comment by asajeffrey** [comment]
Squashed. r? @jdm

File: support/magicleap/Servo2D/code/src/main.cpp [hunk 1]
```diff
@@ -5,6 +5,7 @@
 #include <Servo2D.h>
 #include <ml_lifecycle.h>
 #include <ml_logging.h>
+#include <gst/gst.h>
```
Comment (xclaesse) [inline support/magicleap/Servo2D/code/src/main.cpp]: I think this is useless since you're not using gstreamer in this file.

Comment (asajeffrey) [reply support/magicleap/Servo2D/code/src/main.cpp]: Ah, that's a holdover from the previous code for initializing gstreamer.

File: etc/taskcluster/decision_task.py [hunk 2]
```diff
@@ -876,6 +876,8 @@ def magicleap_build_task(name, build_type):
         .with_early_script("""
             export MAGICLEAP_SDK="$PWD/magicleap/v0.20.0+ndk19c"
             export MLCERT="$PWD/magicleap/TempSharedCert.cert"
+            # gstreamer requires the homebrew bison rather than the system bison
+            export PATH="/usr/local/opt/bison/bin:$PATH"
```
Comment (jdm) [inline etc/taskcluster/decision_task.py]: This is no longer needed, right?

File: etc/taskcluster/macos/Brewfile [hunk 3]
```diff
@@ -1,5 +1,6 @@
 brew "autoconf@2.13"
 brew "automake"
+brew "bison"
```
Comment (jdm) [inline etc/taskcluster/macos/Brewfile]: Same here?

**Review by jdm — CHANGES_REQUESTED** [review]

File: etc/taskcluster/decision_task.py (hunk 2 above)
Comment (asajeffrey) [reply etc/taskcluster/decision_task.py]: Right.

File: etc/taskcluster/macos/Brewfile (hunk 3 above)
Comment (asajeffrey) [reply etc/taskcluster/macos/Brewfile]: Same here.

**Review by jdm — APPROVED** [review]

**Comment by jdm** [comment]
We'll need to wait to merge this until our upstream taskcluster docker issues are resolved.

**Comment by asajeffrey** [comment]
Squashed. This is now ready to land once the CI is well enough.

**Comment by asajeffrey** [comment]
I think that's #9087. @bors-servo retry

**Comment by asajeffrey** [comment]
Hmm, that's #18655, but it's marked as closed? @bors-servo retry

**Comment by jdm** [comment]
The failure reported was actually:
```
{
    "status": "FAIL", 
    "group": "default", 
    "message": "/css/css-transforms/css-transforms-3d-on-anonymous-block-001.html a31aa4b7e008f5f3f2b3c1369e830823183c2789\n/css/css-transforms/reference/css-transforms-3d-anonymous-block-ref.html 7ae3c0e5bd54592e74a2ed814239ef034b85b02a", 
    "stack": null, 
    "subtest": null, 
    "test": "/css/css-transforms/css-transforms-3d-on-anonymous-block-001.html", 
    "line": 31480, 
    "action": "test_result", 
    "expected": "PASS"
}
```
Not sure what you were looking at.

**Comment by asajeffrey** [comment]
Oops, I meant #18865 not #18655.

**Pull request merged**